src/core/DataFile.cpp

```diff
@@ -163,6 +163,53 @@ DataFile::~DataFile()
 
 
 
+bool DataFile::validate( QString extension )
+{
+	switch( m_type )
+	{
+	case Type::SongProject:
+		if( extension == "mmp" || extension == "mmpz" )
+		{
+			return true;
+		}
+		break;
+	case Type::SongProjectTemplate:
+		if(  extension == "mpt" )
+		{
+			return true;
+		}
+		break;
+	case Type::InstrumentTrackSettings:
+		if ( extension == "xpf" || extension == "xml" )
+		{
+			return true;
+		}
+		break;
+	case Type::UnknownType:
+		if (! ( extension == "mmp" || extension == "mpt" || extension == "mmpz" ||
```

Comment:
> we can get back to the pretty select/case statements by creating a FileType enum down the road. I think it will have an added benefit by finally allowing us to remove the mmpz, mpt, xpf hard coded all over the place.


Im not sure I understand the requirements, This code is checking the string of the file extension. Are you suggesting something like the following

````
enum { mmpz, mpt, xpf };
const char *types[] = { "mmpz", "mpt", "xpf" };
.....
.....
.....
if (extension == types[mpt]) .....
````

> It appears we register our plugins as having an extension associated... and I'm guessing the Engine::pluginFileHandling().contains() doesn't actually contain xiz for some reason. I'd guess this is actually an existing bug that surfaced with this code...

The problem with this code as it stands is xiz, are not of Type::UnknownType, the returned enum is undefined. This should not be too hard for me to get sorted :)